packages/scandipwa/src/component/ProductCard/ProductCard.component.js

```diff
@@ -268,29 +268,31 @@ export class ProductCard extends Product {
             inStock
         } = this.props;
 
-        if (inStock && this.requiresConfiguration()) {
-            return (
-                <button
-                  block="Button AddToCart"
-                  mods={ { layout } }
-                  onClick={ showSelectOptionsNotification }
-                >
-                    { __('Add to cart') }
-                </button>
-            );
+        const requiresConfiguration = this.requiresConfiguration();
+
+        if (inStock) {
+            if (requiresConfiguration) {
+                return (
+                    <button
+                      block="Button AddToCart"
+                      mods={ { layout } }
+                      onClick={ showSelectOptionsNotification }
+                    >
+                        { __('Add to cart') }
+                    </button>
+                );
+            }
+
+            return this.renderAddToCartButton(layout);
         }
```

Comment:
Remove please nesting. Can just put blocks if and it will be more readable.
Like

```js

If (inStock && !requiresConfiguration) {
     return this.renderAddToCartButton(layout);
}

If (inStock && requiresConfiguration) {
....
}


```